.github/workflows/pypi-publish.yml

```diff
@@ -5,24 +5,58 @@ name: Publish Python 🐍 distributions 📦 to PyPI
 
 on:
   release:
+    # • `released` reacts to stable releases
+    # • `prereleased` reacts to pre-releases
+    # • `published` reacts to both
     types: [published]
 
 jobs:
-  build-n-publish:
+  build:
+    name: Build distribution 📦
     runs-on: ubuntu-latest
-    permissions:
-      id-token: write # for trusted publishing
 
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-python@v4
+      - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+
+      - uses: actions/setup-python@v5
+
+      - name: Get history and tags for SCM versioning to work
+        run: |
+          git fetch --prune --unshallow
+          git fetch --depth=1 origin +refs/tags/*:refs/tags/*
```

Comment:
Why not pass `fetch-depth` to the checkout action?

Reply:
I was actually checking the options they have, but I was not able to combine it with the first `fetch --prune --unshallow`

Reply:
What's the purpose of that command? If you simply want to do a full clone so that setuptools_scm works properly, you can set fetch-depth to 0. Or is there something else here?